spec/integration/http/client_spec.rb

```diff
@@ -132,9 +132,13 @@
     it "detects when the server has closed the connection and reconnects" do
       https_server.start_server do |port|
         uri = URI("https://127.0.0.1:#{port}")
+        kwargs = {headers: {'Content-Type' => 'text/plain'}, options: {ssl_context: root_context, body: ''}}
 
-        expect(client.get(uri, options: {ssl_context: root_context})).to be_success
-        expect(client.get(uri, options: {ssl_context: root_context})).to be_success
+        expect(client.post(uri, **kwargs)).to be_success
+        # the server closes its connection after each request, so posting
+        # again will force ruby to detect that the remote side closed the
+        # connection, and reconnect
+        expect(client.post(uri, **kwargs)).to be_success
```

Comment:
should we be checking the logs to ensure the new connection is opened?

Reply:
Ah so it's little tricky, because puppet reuses the same `Net::HTTP` connection. But the second time we `post`, ruby detects that the remote side closed the connection and reconnects. The only indication that it does this is a debug message when `http_debug` is enabled. The test is a little fragile, but I don't know of another way to really verify that ruby is doing the right thing.